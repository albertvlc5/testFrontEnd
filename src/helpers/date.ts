export const getPreviousMonthNameByLocale = (
  dateStr: string,
  locale = 'pt-BR',
) => {
  const date = new Date(dateStr);
  const month = date.getUTCMonth();
  const previousMonth = month === 0 ? 11 : month - 1;

  const monthName = new Date(
    date.getUTCFullYear(),
    previousMonth,
    1,
  ).toLocaleString(locale, { month: 'long' });

  return monthName;
};
