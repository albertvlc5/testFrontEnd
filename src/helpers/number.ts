import 'intl';
import 'intl/locale-data/jsonp/pt';
export const formatNumberByLocale = (value: number, locale = 'pt-Br'): string =>
  new Intl.NumberFormat(locale, { minimumFractionDigits: 2 }).format(value);

export const formatMoney2Numeral = (money: string): number => {
  const [value, cents] = String(money).split(',');
  return Number([value.replace(/\D+/g, ''), cents].join('.'));
};

export const numberToCurrencyIntl = (
  value: number = 0,
  locale = 'pt-BR',
  currency = 'BRL',
) =>
  new Intl.NumberFormat(locale, { style: 'currency', currency }).format(value);
