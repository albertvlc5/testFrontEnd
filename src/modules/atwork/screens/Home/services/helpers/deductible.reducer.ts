import {
  IDeductible,
  IDeductibleTransformed,
} from '../../interfaces/deductible.interface';

export const initialDeductibleTransformed = {
  totalDiscounts: 0,
  dueDate: '',
};

export const deductibleReducer = (
  acumulator: IDeductibleTransformed,
  item: IDeductible,
) => ({
  totalDiscounts: acumulator.totalDiscounts + item.amountDeductibleThisMonth,
  dueDate: item.dueDate,
});
