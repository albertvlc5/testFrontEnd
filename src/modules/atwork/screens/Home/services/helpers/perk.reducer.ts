import { ProductType } from '../../interfaces/product-type.enum';
import { IPerk, IPerkTransformed } from '../../interfaces/perk.interface';

export const initialPerkTransformed = {
  totalAmount: 0,
  salaryAdvancement: {
    totalAmountAvailable: 0,
    amountDeductible: 0,
    currency: '',
  },
  payroll: {
    amountDeductible: undefined,
    totalAmountAvailable: 0,
    currency: '',
  },
};

export const perkReducer = (acumulator: IPerkTransformed, item: IPerk) => {
  const salaryAdvancement =
    item.type === ProductType.SALARY_ADVANCEMENT
      ? {
          totalAmountAvailable: item.totalAmountAvailable,
          amountDeductible: item.amountDeductible,
          currency: item.symbol,
        }
      : acumulator.salaryAdvancement;

  const payroll =
    item.type === ProductType.PAYROLL
      ? {
          amountDeductible: item.amountDeductible,
          currency: item.symbol,
          totalAmountAvailable: item.totalAmountAvailable,
        }
      : acumulator.payroll;

  return {
    totalAmount: acumulator.totalAmount + item.totalAmountAvailable,
    salaryAdvancement: salaryAdvancement,
    payroll: payroll,
  };
};
