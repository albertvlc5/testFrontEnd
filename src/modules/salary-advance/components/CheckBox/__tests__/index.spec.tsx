import { fireEvent } from '@testing-library/react-native';
import React from 'react';

import { mountWithAtworkHost } from '../../../../../../src/__tests__/helpers';
import { CheckBox } from '../index';

describe('CheckBox', () => {
  it('should render component', () => {
    const mockOnCheck = jest.fn();
    const wrapper = mountWithAtworkHost(
      <CheckBox value={false} onCheck={mockOnCheck} />,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  describe('When touch the checkbox', () => {
    it('should call onCheck', () => {
      const mockOnCheck = jest.fn();
      const { getByTestId } = mountWithAtworkHost(
        <CheckBox value={false} onCheck={mockOnCheck} />,
      );
      const comp = getByTestId('box-id');
      fireEvent(comp, 'onTouchEnd');
      expect(mockOnCheck).toHaveBeenCalled();
    });
  });
});
