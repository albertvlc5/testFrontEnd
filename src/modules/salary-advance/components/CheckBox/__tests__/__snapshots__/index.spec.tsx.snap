// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should render component 1`] = `
<View
  onTouchEnd={[MockFunction]}
  style={
    Array [
      Object {
        "borderColor": "#111515",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 30,
        "paddingBottom": 3,
        "paddingLeft": 3,
        "paddingRight": 3,
        "paddingTop": 3,
        "width": 30,
      },
    ]
  }
  testID="box-id"
>
  <View
    checked={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 5,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="check-value-id"
  />
</View>
`;
